qa-test-app/src/main/java/com/mapbox/navigation/qa_test_app/view/customnavview/MapboxNavigationViewCustomizedActivity.kt

```diff
@@ -858,45 +857,43 @@ class MapboxNavigationViewCustomizedActivity : DrawerActivity() {
         }
     }
 
-    private val backPressOverride = object : NavigationViewListener() {
+    private val navigationStateListener = object : NavigationViewListener() {
 
         override fun onFreeDrive() {
-            lastNavigationState = NavigationState.FreeDrive
+            binding.navigationView.registerMapObserver(onMapLongClick)
+            onBackPressCallback.isEnabled = false
         }
 
         override fun onDestinationPreview() {
-            lastNavigationState = NavigationState.DestinationPreview
+            binding.navigationView.unregisterMapObserver(onMapLongClick)
+            onBackPressCallback.isEnabled = false
         }
 
         override fun onRoutePreview() {
-            lastNavigationState = NavigationState.RoutePreview
+            binding.navigationView.unregisterMapObserver(onMapLongClick)
+            onBackPressCallback.isEnabled = true
         }
 
         override fun onActiveNavigation() {
-            lastNavigationState = NavigationState.ActiveNavigation
+            binding.navigationView.unregisterMapObserver(onMapLongClick)
+            onBackPressCallback.isEnabled = false
         }
 
         override fun onArrival() {
-            lastNavigationState = NavigationState.Arrival
+            binding.navigationView.unregisterMapObserver(onMapLongClick)
+            onBackPressCallback.isEnabled = false
         }
+    }
 
-        override fun onBackPressed(): Boolean {
```

Comment:
This api also gives you the ability to customize when back press is handled right? Removing it, removes that ability. And now it errors inside when back pressed is handled in the FreeDrive state. 

It seems we could leave this function so there is a way to control the back press

Reply:
But how could you override the behavior of `NavigationView`?

Reply:
Yeah it is easy to add your own back press. I like that.

I'm saying, `onBackPressed(): Boolean` returns a boolean. It allowed you to disable `NavigationView` back press handling.

For example, you can disable the "go back during active guidance"

Reply:
OH nevermind I see

This dispatcher is only enabled for the onRoutePreview callback 🤔 
```
        override fun onRoutePreview() {
            binding.navigationView.unregisterMapObserver(onMapLongClick)
            onBackPressCallback.isEnabled = true
        }
```

Hmm. It works, makes me wonder if it is more difficult to manage the state that way. But maybe it's just different

Reply:
With this custom callback implementation, this example activity is adding it's own `onBackPressedCallback`. However, the `backpressCallback` added here
```
fun attach(
            componentActivity: ComponentActivity,
            store: Store,
            lifecycleOwner: LifecycleOwner,
        ) {
            val onBackPressedCallback = DefaultOnBackPressedCallback(store)
            componentActivity.onBackPressedDispatcher.addCallback(onBackPressedCallback)
```
still remains. I am wondering if we need to expose something that would allow users to remove the `DefaultOnBackPressedCallback` 🤔 Otherwise, we will have multiple `onBackPressedCallback` in this specific case:
- `DefaultOnBackPressedCallback` added by `NavigationView`.
- `MyCustomOnBackPressedCallback` added by example activity.

Reply:
@abhishek1508 only one callback will process the back press event. `OnBackPressedDispatcher` selects an enabled callback that was added last. If there are no enabled callbacks at the moment, then default behavior will happen (e.g. current activity will finish).